src/System.Text.Json/src/System/Text/Json/Serialization/JsonSnakeCaseNamingPolicy.cs

```diff
@@ -0,0 +1,67 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Globalization;
+
+namespace System.Text.Json
+{
+    internal sealed class JsonSnakeCaseNamingPolicy : JsonNamingPolicy
+    {
+        public override string ConvertName(string name)
+        {
+            if (string.IsNullOrEmpty(name))
+                return name;
+
+            StringBuilder builder = new StringBuilder(name.Length + Math.Min(2, name.Length / 5));
```

Comment:
Where did the `Math.Min(2, name.Length / 5)` formula come from?  A comment would be helpful.

Reply:
Oh, there is a typo here. There should be `Max` instead of `Min`. Fixed and comment added.